rotkehlchen/db/dbhandler.py

```diff
@@ -1973,6 +1977,151 @@ def delete_trade(self, trade_id: str) -> Tuple[bool, str]:
         self.conn.commit()
         return True, ''
 
+    def add_amm_trades(self, trades: List[AMMTrade]) -> None:
+        trade_tuples: List[Tuple[Any, ...]] = []
+        for trade in trades:
+            trade_tuples.append(trade.to_db_tuple())
+
+        query = (
+            """
+            INSERT INTO amm_trades (
+                tx_hash,
+                log_index,
+                address,
+                timestamp,
+                location,
+                type,
+                is_base_asset_unknown,
+                base_asset_address,
+                base_asset_symbol,
+                base_asset_name,
+                base_asset_decimals,
+                is_quote_asset_unknown,
+                quote_asset_address,
+                quote_asset_symbol,
+                quote_asset_name,
+                quote_asset_decimals,
+                amount,
+                rate,
+                fee,
+                notes
+            )
+            VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
+            """
+        )
+        self.write_tuples(tuple_type='amm_trade', query=query, tuples=trade_tuples)
+
+    def edit_amm_trade(self, trade: AMMTrade) -> Tuple[bool, str]:
```

Comment:
You can't edit an on-chain trade :sweat_smile: 

Editing is only allowed for manual trades that the user has added on their own

Reply:
@vnavascues For stuff like that please ask me if you are not sure. I hoped it would be obvious you can't edit an an on-chain trade. But please always ask as early as possible instead of wasting time in implementing something we will delete.

Reply:
It was easy to implement and I was busy with other stuff. I didn't realise until I was about to deliver this that it clicked... editing on-chain trades? mmm... this is why the comment, sorry. But what about editing the `notes` column?

Reply:
We can indeed add notes to trades theoretically but we don't allow this for anything other than manual trades at the moment.